### PR TITLE
Modify day summary prompt to define a structure

### LINKED ISSUE #1057: Day summary message formatting
have a fixed structure returned from the backend, force gpt to follow a specific structure
Image

### PR DESCRIPTION
**Pull request by mdmohsin7**
Closes #1057 
Screenshot 2024-10-13 at 12 47 36 PM

IMG_0515

## Summary by CodeRabbit

- **New Features**
	- Enhanced memory processing and context retrieval functionalities.
	- Introduced constraints for concise output in memory summaries.
	- Updated prompts for better date context retrieval and key point extraction from conversations.

- **Bug Fixes**
	- Improved handling of responses with minimal content to prevent errors.

- **Refactor**
	- Streamlined output structures for experience summaries and fact extraction processes.
	- Clarified criteria for identifying new user facts.

### CODE REVIEW

File: backend/utils/llm.py [hunk 1]
```diff
@@ -178,7 +178,8 @@ def get_memory_summary(uid: str, memories: List[Memory]) -> str:
     Remember {user_name} is busy so this has to be very efficient and concise.
     Respond in at most 50 words.
   
-    Output your response in plain text, without markdown.
+    Output your response in plain text, without markdown. No newline character and only use numbers for the action items.
+    The total words in the response should not exceed 50 words.
```
Comment (beastoin) [inline backend/utils/llm.py]: we have the limit at 179, should we remind llm about the max numbers of words?

File: backend/utils/llm.py [hunk 2]
```diff
@@ -178,7 +178,8 @@ def get_memory_summary(uid: str, memories: List[Memory]) -> str:
     Remember {user_name} is busy so this has to be very efficient and concise.
     Respond in at most 50 words.
   
-    Output your response in plain text, without markdown.
+    Output your response in plain text, without markdown. No newline character and only use numbers for the action items.
```
Comment (beastoin) [inline backend/utils/llm.py:181]: I understand that we need. to fix the daily memory summary right? is 'get_memory_summary' func only used by the daily memory summary feature?

File: backend/utils/llm.py (hunk 1 above)
Comment (mdmohsin7) [reply backend/utils/llm.py]: It's upto us, I added at the end again to enforce the limit. Should I remove it?

File: backend/utils/llm.py (hunk 2 above)
Comment (mdmohsin7) [reply backend/utils/llm.py:181]: Yes, I only found one reference for `get_memory_summary ` in the codebase

File: backend/utils/llm.py (hunk 1 above)
Comment (beastoin) [reply backend/utils/llm.py]: keep the prompt simple as much as posible, try removing it and see what llm give us man.

Comment (mdmohsin7) [reply backend/utils/llm.py]: Done! Gives the same response

**Pull request merged**